typeshed/__init__.py

```diff
@@ -0,0 +1,17 @@
+import os
+
+
+__all__ = ['__location__', '__version__']
+
+
+try:
+    import pkg_resources
+except ImportError:
+    __location__ = os.path.dirname(os.path.dirname(__file__))
+else:
+    __location__ = pkg_resources.resource_filename("typeshed", "")
```

Comment:
What is the significance of `__location__`?  I don't recall the docs and neither does Google.

Reply:
No significance, just like `__version__`. It's just a name for people to use to figure out where their stubs live:
```py3
import typeshed

subprocess.run(["mypy", f"--custom-typeshed-dir={typeshed.__location__}"])
```

I didn't want to use `__path__` because that one *does* change where importlib would look for sub-packages and that would break for the `pkg_resources` version which unpacks typeshed to a temporary directory.

Do you have a different suggestion for this name?

Reply:
Ah. That's why comments are useful. :-)

I'd avoid using a dunder name, since that suggests a connection with official Python stuff. (Technically *all* dunder names are reserved by Python.)

I'd name this `typeshed_location` or `typeshed_dir` (the latter matching `--custom-typeshed-dir` more closely). You can even make it a function if you want to be fancy -- that makes it easier to turn it into something algorithmically complex.

Also, will the fancy `resource_filename()` call actually work if, say, typeshed was installed as a zip file? Otherwise is there a reason not to just use the fallback based on `__file__` always?

Reply:
> (Technically *all* dunder names are reserved by Python.)

True but that ship has long sailed for metadata like `__version__`, `__author__`, etc.  The standard library alone has 67 and 31 instances of those, respectively. I wanted `__location__` to be consistent with `__version__` but you're right that it's probably not worth it as it might be confusing.

Since we are already importing typeshed, including "typeshed" in the name of the variable feels icky to me. I'll just make a function called `get_dir()`.

----
> Also, will the fancy `resource_filename()` call actually work?

Yes, [the pkg_resources docs state](https://setuptools.readthedocs.io/en/latest/pkg_resources.html#resource-extraction) that:
>  If the resource is in an archive distribution (such as a zipped egg), it will be extracted to a cache directory, and the filename within the cache will be returned.

I tested this and it works. This is why I set `zip_safe` to True in setup.py. The good thing about making `get_dir()` a function is that unless called, the import alone won't extract typeshed to a temporary directory in this situation.

Reply:
OK, get_dir() sounds fine. (And as you saw `__version__` and `__author__` have some kind of semi-official status, though I can't find a PEP -- probably the convention predates the PEP system.)

Happy that `pkg_resources` is thorough!